js/src/personal/haoking/projects/RightProjects.tsx

```diff
@@ -1,52 +1,112 @@
-import { Text, Title, Box, Divider, Badge, Group } from '@mantine/core'
+import {
+  Text,
+  Title,
+  Box,
+  Divider,
+  Badge,
+  Group,
+  useComputedColorScheme,
+} from '@mantine/core'
+import { useHover } from '@mantine/hooks'
 import styles from '../Haoking.module.css'
 import { theme } from '../theme.ts'
 
 /* Component for right side of Projects page */
-export function RightProjects() {
+export function RightProjects({ setImage }: { setImage: any }) {
+  const colorScheme = useComputedColorScheme()
+  const { hovered: hovered1, ref: ref1 } = useHover()
+  const { hovered: hovered2, ref: ref2 } = useHover()
+  const { hovered: hovered3, ref: ref3 } = useHover()
+  const projectsData = [
+    {
+      title: 'Todo App',
+      subtitle: 'manage your tasks',
+      link: 'https://github.com/luohk000/todo-app',
+      image:
+        'https://i.natgeofe.com/k/6289c775-a06c-426a-badb-8d181a55237b/raccoon-grass_2x1.jpg',
+      tools: ['flutter'],
+      isHovered: hovered1,
+      ref: ref1,
+    },
+    {
+      title: 'Quiz App',
+      subtitle: 'quiz yourself',
+      link: 'https://github.com/luohk000/quizWIP',
+      image:
+        'https://m.media-amazon.com/images/I/71HJRIGU81L._AC_UF894,1000_QL80_.jpg',
+      tools: ['flutter', 'firebase'],
+      isHovered: hovered2,
+      ref: ref2,
+    },
+    {
+      title: 'Word Search Generator',
+      subtitle: 'fun stuff',
+      link: '',
+      image:
+        'https://www.westernpest.com/wp-content/uploads/Raccoon_in_Trash_iStock-91705369-scaled.jpg',
+      tools: ['c++'],
+      isHovered: hovered3,
+      ref: ref3,
+    },
+  ]
+
+  let anyHovered = false
+  projectsData.forEach((project) => {
+    if (project.isHovered) {
+      setImage(project.image)
+      anyHovered = true
+    }
+  })
+  if (!anyHovered) setImage('')
+
   return (
     <Box
       ml={90}
-      mr={350}
+      mr={300}
       mt={45}
-      maw={400}
+      maw={600}
       className={styles.animate_fade_in_delay}
     >
       <Divider
+        color={colorScheme === 'light' ? '#000' : '#fff'}
         mt={120}
         pt={20}
         maw={350}
-        color={theme.colors.white[6]}
       />
-      <Group>
-        <Title c={theme.colors.white[1]}>
-          <Text
-            inherit
-            component="a"
-            href="https://github.com/luohk000/todo-app"
-            target="_blank"
-          >
-            {'Todo App'}
-          </Text>
-        </Title>
-        <Badge color={theme.colors.black[6]}>{'Flutter'}</Badge>
-      </Group>
-      <Text c={theme.colors.white[2]}>{'manage your tasks'}</Text>
-      <Group>
-        <Title c={theme.colors.white[1]}>
-          <Text
-            inherit
-            component="a"
-            href="https://github.com/luohk000/quizWIP"
-            target="_blank"
+      {projectsData.map((project) => (
```

Comment:
if you refactor this out into its own component, then you can each of them use `useHover` and pass in `setImage()` to avoid hard coding useHover 3 times.